Source/lib/pdf417/encoder/PDF417.cs

```diff
@@ -722,13 +724,36 @@ internal void generateBarcodeLogic(String msg, int errorCorrectionLevel)
       /// </summary>
       /// <param name="sourceCodeWords">number of code words</param>
       /// <param name="errorCorrectionCodeWords">number of error correction code words</param>
+      /// <param name="longDimension">The longest dimension of the barcode, used for columns</param>
+      /// <param name="shortDimension">The short dimension of the barcode, used for rows</param>
+      /// <param name="aspectRatio">The height of a row, will alter this parameter if aspectRatio>4 (aspectRatio==AUTO)</param>
       /// <returns>dimension object containing cols as width and rows as height</returns>
-      private int[] determineDimensions(int sourceCodeWords, int errorCorrectionCodeWords)
+      private int[] determineDimensions(int sourceCodeWords, int errorCorrectionCodeWords, int longDimension, int shortDimension, ref int aspectRatio)
       {
          float ratio = 0.0f;
          int[] dimension = null;
 
-         for (int cols = minCols; cols <= maxCols; cols++)
+         int dimMaxCols = Math.Min((longDimension- START_STOP_WIDTH) / COLUMN_WIDTH, maxCols);
+         int dimMaxRows = Math.Min(shortDimension/aspectRatio, maxRows);
+         int calculatedRows = calculateNumberOfRows(sourceCodeWords, errorCorrectionCodeWords, dimMaxCols);
+         bool canFit = calculatedRows <= dimMaxRows;
+
+         //Set the aspectRatio if AUTO.
+         if (aspectRatio>=(int)PDF417AspectRatio.AUTO)
+         {
+            dimMaxRows = Math.Min(shortDimension, maxRows);
+            int newRatio = 4;
+                
+            //Integer division.
+            if (dimMaxRows >= calculatedRows)
+                newRatio = Math.Min(dimMaxRows / calculatedRows, newRatio);
+            aspectRatio = newRatio;
+            dimMaxRows = Math.Min(shortDimension/aspectRatio, maxRows);

```

Comment:
Throws a division by 0 if I try to encode the value "2345" with the following options:
AspectRatio = AUTO
ErrorCorrection = AUTO
Height = 1
Width = 1
Can you confirm the exception and fix this please?